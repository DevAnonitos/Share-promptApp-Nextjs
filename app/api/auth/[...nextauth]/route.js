import NextAuth from "next-auth/next";
import GoogleProvider from "next-auth/providers/google";

const handler = NextAuth({
    providers: [
        GoogleProvider({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET,
        })
    ],
    callbacks: {
        async session({ session }) {

        },
        async signIn({ account, profile, user, credentials }) {
            try {

            } catch (error) {
                console.log("Error checking if user exists: ", error.message);
                return false;
            }
        },
    }
});

export {
    handler as GET,
    handler as POST,
};
