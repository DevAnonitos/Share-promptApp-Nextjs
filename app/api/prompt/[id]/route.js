import Prompt from "../../../../models/prompt";
import { connectToDB } from "../../../../utils/database";

export const GET = async (request, { params }) => {
    try {
        await connectToDB();

        const prompt = await Prompt.findById(params.id).populate("creator");

        if(!prompt) {
            return new Response("Prompt not found", {
                status: 404,
            });
        }

        return new Response(JSON.stringify(prompt), {
            status: 200,
        });
    } catch (error) {
        return new Response("Internal Server Error", {
            status: 500,
        });
    }
};

export const PATH = async (request, { params }) => {
    const { prompt, tag } = await request.json();

    try {
        await connectToDB();

        const existingPrompt = await Prompt.findById(params.id);

        if(!existingPrompt) {
            return new Response("Prompt not founded", {
                status: 404,
            });
        }

        existingPrompt.prompt = prompt;
        existingPrompt.tag = tag;

        return new Response("Successfully update-prompt!", {
            status: 200,
        });

    } catch (error) {
        return new Response("Error updating Prompt", {
            status: 500,
        });
    }
};

export const DELETE = async (request, { params }) => {
    try {
        await connectToDB();

        await Prompt.findByIdAndRemove(params.id);

        return new Response("Prompt delete successfully", {
            status: 200,
        });
    } catch (error) {
        return new Response("Error deleting prompt", {
            status: 500,
        });
    }
};
