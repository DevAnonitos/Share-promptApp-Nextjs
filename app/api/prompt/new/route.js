import { connectToDB } from "@/utils/database";

export const POST = async (request) => {
    const { userId, prompt, tag } = await request.json();

    try {
        await connectToDB();

        return new Response(JSON.stringify(), {
            status: 201,
        })
    } catch (error) {
        return new Response("Fail to create a new prompt", {
            status: 500,
        })
    }
};
