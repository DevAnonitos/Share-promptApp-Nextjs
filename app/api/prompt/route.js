import Prompt from "@/models/prompt";
import { connectToDB } from "@/libs/utils/database";

export const GET = async (request) => {
    try {
        await connectToDB();

        const prompts = await Prompt.find({}).populate('creator');

        return new Response(JSON.stringify(prompts), {
            status: 200,
        });
    } catch (error) {
        return new Response(JSON("Fail to fetch All prompts", {
            status: 500,
        }))
    }
};
