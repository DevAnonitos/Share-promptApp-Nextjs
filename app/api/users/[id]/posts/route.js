import Prompt from "@/libs/models/prompt";
import { connectToDB } from "@/libs/utils/database";

export const GET = async (request, { params }) => {
    try {
        await connectToDB();

        if(!params || !params.id) {
            return new Response("UserId is missing", {
                status: 400,
            });
        }

        const prompts = await Prompt.find({ creator: params.id }).populate("creator");

        return new Response(JSON.stringify(prompts), {
            status: 200,
        });
    } catch (error) {
        console.log(error);
        return new Response("Failed to fetch prompts created by user", {
            status: 500
        });
    }
};
