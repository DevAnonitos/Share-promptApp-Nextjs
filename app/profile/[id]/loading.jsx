import React from 'react';
import Image from 'next/image';

// Create Loading Components
const Loading = () => {
    return (
        <div className='w-full flex-center'>
            <Image
                src='assets/icons/loader.svg'
                width={50}
                height={50}
                alt='loader'
                className='object-contain'
            />
        </div>
    );
};

export default Loading;
