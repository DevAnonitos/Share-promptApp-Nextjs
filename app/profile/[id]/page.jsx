"use client"

import React, { useState, useEffect } from 'react';
import { useSearchParams } from 'next/navigation';
import { Profile } from '@/components';

const UserProfile = ({ params }) => {
    const searchParams = useSearchParams();
    const userName = searchParams.get("username");

    const [userPosts, setUserPosts] = useState([]);

    useEffect(() => {
        const fetchPosts = async () => {
            try {
                const response = await fetch(`/api/users/${params?.id}/posts`, {
                    next: { revalidate: 60 },
                });
                const data = await response.json();
    
                setUserPosts(data);
            } catch (error) {
                console.error("Error fetching posts:", error);
            }
        };

        if (params?.id) fetchPosts();
    }, [params.id]);

    return (
        <Profile
            name={userName}
            desc={`Welcome to ${userName}'s personalized profile page. Explore ${userName}'s exceptional prompts and be inspired by the power of their imagination`}
            data={userPosts}
        />
    );
};

export default UserProfile;
