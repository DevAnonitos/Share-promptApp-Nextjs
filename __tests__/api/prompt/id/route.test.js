/**
* @jest-environment node
*/
import { GET } from "@/app/api/prompt/[id]/route";

describe("API prompt ID", () => {
  const apiUrl = "http://localhost:3000/api/prompt/645cef4b3a66a9ea8f5f46c9"
  it("should return data json from API route Prompt ID", async () => {
    const res = await fetch(apiUrl);
    const data = await res.json();
    console.log(data);
    expect(data).toBeDefined()
    expect(data).toMatchSnapshot()
  })

  it("should return status 200 from API route Prompt ID", async () => {
    const res = await fetch(apiUrl);
    const data = await res.json();
    console.log(data);
    expect(res.status).toBe(200)
    expect(data).toMatchSnapshot()
  })
})