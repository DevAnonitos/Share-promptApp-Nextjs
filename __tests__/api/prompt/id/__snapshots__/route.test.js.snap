// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API prompt ID should return data json from API route Prompt ID 1`] = `
{
  "__v": 0,
  "_id": "645cef4b3a66a9ea8f5f46c9",
  "creator": {
    "__v": 0,
    "_id": "645a5ff09c6a32502c433959",
    "email": "nguyebao281003@gmail.com",
    "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
    "username": "nguyebao281003",
  },
  "prompt": "Brad Rigney is a concept artist of fantasy themes, 4 : 3",
  "tag": "stable-diffusion",
}
`;

exports[`API prompt ID should return status 200 from API route Prompt ID 1`] = `
{
  "__v": 0,
  "_id": "645cef4b3a66a9ea8f5f46c9",
  "creator": {
    "__v": 0,
    "_id": "645a5ff09c6a32502c433959",
    "email": "nguyebao281003@gmail.com",
    "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
    "username": "nguyebao281003",
  },
  "prompt": "Brad Rigney is a concept artist of fantasy themes, 4 : 3",
  "tag": "stable-diffusion",
}
`;
